backend/fms_core/services/study.py

```diff
@@ -80,3 +99,19 @@ def get_study(project_obj: Project, study_letter: str):
         errors.append(f"Both a project and a study letter are required to retrieve a study.")
 
     return study, errors, warnings
+
+def can_remove_study(study: int) -> Tuple[bool, Dict[str, List[str]], Dict[str, List[str]]]:
+    warnings = {}
+    errors = defaultdict(list)
+    is_removable = True
+
+    if StepHistory.objects.filter(study=study).exists():
```

Comment:
filter using study_id=study_id instead of study=study_id... I am not convinced an instance of study can be equated to an int in django. This will not fail it will just always be false.

Reply:
you sure it's not two underscores?

Reply:
two underscore means it follows the FK to the next model, but django as a strange way of defining FK it creates an attribute by appending an "_id" after the name you gave the FK field. So the study fk fields has a study_id attribute which does not require following the relation. It is equivalent, but i am not sure Django does not query the foreign model if you use the "__" notation.